kancolle_auto.sikuli/combat.sikuli/combat.py

```diff
@@ -490,20 +491,62 @@ def go_repair(self):
+                            if global_regions['repair_panel'].exists(Pattern('ship_class_ss.png').similar(0.8)) or global_regions['repair_panel'].exists(Pattern('ship_class_ssv.png').similar(0.65)):
+                                # Ship is submarine
+                                # Try setting next sortie time according to repair timer
+                                log_msg("Submarine denied bucket. ;_;")
+                                timer = self.timer_end(int(repair_timer[0:2]), int(repair_timer[3:5]) - 1)
+                                log_success("Repair should be done at %s" % timer.strftime("%Y-%m-%d %H:%M:%S"))
+                                self.next_sortie_time_set(int(repair_timer[0:2]), int(repair_timer[3:5]) - 1)
+                                self.repair_timers.append(timer)
+                                self.repair_timers.sort()
+                                empty_docks -= 1
+                            else:
+                                # Ship is not submarine
+                                log_msg("Not submarine!")
+                                log_success("Using bucket for all repairs!")
+                                self.kc_region.click('repair_bucket_switch.png')
+                                self.next_sortie_time_set(0, 0)
+                                bucket_use = True
                     else:
                         # Otherwise, act accordingly to timer and repair timer limit
                         repair_timer = check_timer(self.kc_region, 'repair_timer.png', 'r', 80, 5)
                         if int(repair_timer[0:2] + repair_timer[3:5]) >= self.repair_time_limit:
                             # Use bucket if the repair time is longer than desired
-                            log_success("Repair time too long... using bucket!")
-                            self.kc_region.click('repair_bucket_switch.png')
-                            self.next_sortie_time_set(0, 0)
-                            bucket_use = True
+                            if (not self.submarine_switch) or (self.submarine_switch and self.submarine_switch_use_buckets):
+                                # Use buckets on subs if submarine handling is disabled, or buckets use is allowed
+                                log_success("Repair time too long... using bucket!")
+                                self.kc_region.click('repair_bucket_switch.png')
+                                self.next_sortie_time_set(0, 0)
+                                bucket_use = True
+                            else:                                
+                                # Submarine switch is enabled, but bucket use is disallowed                                
+                                if global_regions['repair_panel'].exists(Pattern('ship_class_ss.png').similar(0.8)) or global_regions['repair_panel'].exists(Pattern('ship_class_ssv.png').similar(0.65)):
```

Comment:
There's a `CLASS_SIMILARITY` constant defined at the beginning of the file. Was this not sufficient for `ship_class_ss` and `ship_class_ssv`? I'd like to set the threshold for all ship class matching to a constant if at all possible.